Added settings panel

brzep opened the pull request:
issue https://github.com/software-mansion/live-compositor/issues/642

- added modal allowing for choice of each input's resolution
- changed render request structure to also send inputs info

wkozyra95 commented on docs/src/components/PlaygroundRenderSettingsInputs.tsx:
InputSettings

wkozyra95 commented on docs/src/components/PlaygroundRenderSettingsInputs.tsx:
```diff
@@ -0,0 +1,119 @@
+  closeModal,
+  handleSettingsUpdate,
+}: PlaygroundSettingsInputsProps) {
+  const [selectedValues, setSelectedValues] = useState<InputResolutionsNames>({
```
- selectedValues is meaningless, those are inputs resolutions or inputs settings
- you don't need this state, by duplicating this logic you need to make sure that this state and playground state are initialized to the same value. It is easier to just pass the state from the above

wkozyra95 commented on docs/src/components/PlaygroundRenderSettingsInputs.tsx:
```diff
@@ -0,0 +1,119 @@
+      ...selectedValues,
+      [inputId]: event.target.value,
+    });
+    handleSettingsUpdate(inputId, nameToResolution(event.target.value));
```
You don't have to map the resolution here, you can do that when building request body

wkozyra95 commented on docs/src/pages/playground.tsx:
```diff
@@ -1,21 +1,88 @@
+  ],
+};
+
+const INITIAL_INPUTS_RESOLUTIONS: InputResolutions = {
```
If it's just used in one place I would inline this

wkozyra95 commented on docs/src/components/PlaygroundRenderSettingsInputs.tsx:
```diff
@@ -0,0 +1,119 @@
+  }
+
+  return (
+    <ReactModal
+      isOpen={isOpen}
```
It's fine, but I would expect component called Input setting to be responsible only for popup content and parent component would include the actual ReactModal

wkozyra95 commented on docs/src/components/PlaygroundRenderSettings.tsx:
```diff
@@ -1,31 +1,36 @@
+          style={{ margin: '10px' }}>
+          Inputs settings
+        </button>
+        <PlaygroundRenderSettingsInputs
```
If this is something that has to somwhere in the component tree, but it does not matter where then put it e.g. on as the last item in root div?

wkozyra95 commented on docs/src/components/PlaygroundRenderSettings.tsx:
```diff
@@ -1,31 +1,36 @@
-            Submit
-          </button>
-        </div>
+      <div>
```
why?

WojciechBarczynski reviewed (commented):
It's not responsive. It doesn't have to look perfect on mobile, but it
 shouldn't break.

PS When creating frontend PR, it's helpful to attach the screenshot of the result to PR's description. It allows the reviewer to validate the graphical design without launching the code.

<img width="493" alt="Screenshot 2024-08-12 at 10 08 30" src="https://github.com/user-attachments/assets/200a138a-3643-4d3e-a380-069b4e503c76">

wkozyra95 commented on docs/src/components/PlaygroundRenderSettings.module.css:
```diff
@@ -0,0 +1,22 @@
+}
+
+.content {
+    zoom: 1.28;
```
```suggestion
    zoom: 1.28;
```

wkozyra95 commented on docs/src/components/SettingsInputs.module.css:
PlaygroundRenderSettings.tsx -> PlaygroundSettings.tsx
SettingInputs -> PlaygroundInputSettings

the same for component names

wkozyra95 commented on docs/src/components/SettingsInputs.module.css:
```diff
@@ -0,0 +1,37 @@
+.container {
```
this layouts need to changes, but as we discussed it can land in follow up

wkozyra95 commented on docs/src/components/SettingsInputs.tsx:
```diff
@@ -0,0 +1,64 @@
+interface PlaygroundSettingsInputsProps {
+  closeModal: () => void;
+  handleSettingsUpdate: (string, Resolution) => void;
+  selectedInputResolutionNames: InputResolutionNames;
```
```suggestion
  inputResolutions: InputResolutionNames;
```
or
```suggestion
  inputs: InputResolutionNames;
```

wkozyra95 commented on docs/src/Resolution.tsx:
Resolution.tsx -> resolutions.ts

tsx is for components
Upper case file are usually named like that after the component, for files with regular js lower case is more standart

brzep commented:
- changed layout of whole playground settings pane
  - ready for shaders and images modal as well as output resolution settings, but disabled until available
- added tooltip to input_id in inputs resolutions modal
- moved `SubmitButton` to `PlaygroundSettings`
- resized footer

full playground:
<img width="1918" alt="image" src="https://github.com/user-attachments/assets/f3dbb06e-c713-4688-88c8-b9a2e3b50572">

on mobile:
<img width="452" alt="image" src="https://github.com/user-attachments/assets/bf523ef3-3436-468b-ac83-6e4da3ad7d58">

with unavailable options disabled:
<img width="1919" alt="image" src="https://github.com/user-attachments/assets/4863a811-497e-42cb-a153-78985afc2f24">

input_id tooltip
<img width="888" alt="image" src="https://github.com/user-attachments/assets/2959d7f9-9129-49e9-9065-cc32173ffd84">

wkozyra95 commented on docs/src/resolution.ts:
```diff
@@ -0,0 +1,88 @@
+  Resoultion1080x1440 = 'Resoultion1080x1440',
+}
+
+export type InputResolutionNames = {
```
```suggestion
export type InputResolutions = {
```
or 
```suggestion
export type InputsSettings = {
```

wkozyra95 commented on docs/src/components/PlaygroundSettings.module.css:
```diff
@@ -0,0 +1,147 @@
+
+.card {
+  min-width: 11rem;
+  width: calc((100% - 3 * (2 * 10px))/ 3 );
```
this looks scary

wkozyra95 commented on docs/src/components/PlaygroundSettings.module.css:
```diff
@@ -0,0 +1,147 @@
+}
+
+.cardTitle {
+  font-size: 1.1rem;
```
use pixels for font size

wkozyra95 reviewed (commented):
did not review much of the css side, this will need to be reworked later, but I don't think this should be a blocker before merging this PR

brzep commented:
new settings layout
<img width="1920" alt="image" src="https://github.com/user-attachments/assets/9d95d7b3-8ce7-498a-90f3-38e7439ec1ac">

wkozyra95 commented on docs/src/pages/playground.module.css:
```diff
@@ -2,7 +2,7 @@
   flex-wrap: wrap;
-  height: 75vh;
+  min-height: 82vh;
   margin: 20px;
```
```suggestion
  margin: 20px;
  flex: 1;
```
to remove that empty space under
